File: Robotics/python/prime.py
class prime:
    def __init__(self, n):
        self.n = n
        print(self.check(n))
        
    def check(self, n):
        if n < 2:
            return False
        if (n % 2 == 0) and (n != 2):
            return False
        if (n % 3 == 0) and (n != 3):
            return False
        i = 5
        w = 2
        while i * i <= n:
            if n % i == 0:
                return False
            i += w
            w = 6-w # formula
        return True

p = prime(30)

File: Robotics/java/Prime.java
package Robotics.java;

public class Prime {

    private int n;

    public Prime(int n) {
        this.n = n;
        System.out.println(check(n));
    }

    private boolean check(int n) {
        if (n < 2) {
            return false;
        }
        if (n % 2 == 0 && n != 2) {
            return false;
        }
        if (n % 3 == 0 && n != 3) {
            return false;
        }
        int i = 5;
        int w = 2;
        while (i * i <= n) {
            if (n % i == 0) {
                return false;
            }
            i += w;
            w = 6 - w;
        }
        return true;
    }

    public static void main(String[] args) {
        Prime p = new Prime(29);
    }
}
